cmd/gitops/beta/run/cmd.go

```diff
@@ -50,6 +74,266 @@ func betaRunCommandPreRunE(endpoint *string) func(*cobra.Command, []string) erro
+			fmt.Println("error getting list:", err)
+		} else {
+			for _, item := range listResult.Items {
+				if item.GetLabels()[flux.VersionLabelKey] != "" {
+					u = item
+					break
+				}
+			}
+		}
+
+		fluxVersion, err := getFluxVersion(u)
+		if err != nil {
+			fmt.Println(err)
+
+			fluxVersion = ""
+		}
+
+		fmt.Println("fluxVersion:", fluxVersion)
+
+		// If Flux is not installed on the cluster then the prerequisites will be installed to initiate the reconciliation process.
+		// This includes all default controllers to set up a reconciliation loop such as the notification-controller, helm-controller, kustomization-controller, and source-controller.
+		// This will also add all relevant CRDs from the controllers above such as Kustomizations, Helm Releases, Git Repository, Helm Repository, Bucket, Alerts, Providers, and Receivers.
+
+		if fluxVersion == "" {
+			filePath := args[0]
+
+			err = installFlux(kubeClient, ctx2, filepath.Join(filePath, fluxDirectory), shortFilename)
+			if err != nil {
+				return fmt.Errorf("flux installation failed: %w", err)
+			}
+		}
+
+		// If Flux is installed on the cluster then we do not need to install flux.
+		// ^^^ this should be easy! :-)
+
+		// Note:
+		// This should be able to work with local (kind and k3d) and remote clusters.
+
+		// Out of scope:
+		// This just includes installing Flux onto the cluster but does not involve creating the prerequisite reconciliation.
+		// This does not include image auto CRDs and controller from Flux. These are not normally installed by default and will be dealt with in a future story.
+
 		return nil
 	}
 }
+
+func getFluxVersion(obj unstructured.Unstructured) (string, error) {
+	labels := obj.GetLabels()
+	if labels == nil {
+		return "", fmt.Errorf("error getting labels")
+	}
+
+	fluxVersion := labels[flux.VersionLabelKey]
+	if fluxVersion == "" {
+		return "", fmt.Errorf("error getting flux version")
+	}
+
+	return fluxVersion, nil
+}
+
+func installFlux(kubeClient ctrlclient.Client, ctx context.Context, filePath string, shortFilename string) error {
+	fmt.Println("filepath:", filePath)
+	fmt.Println("shortFilename:", shortFilename)
+
+	opts := install.Options{
+		BaseURL:      install.MakeDefaultOptions().BaseURL,
+		Version:      "v0.31.2",
+		Namespace:    "flux-system",
+		Components:   []string{"source-controller", "kustomize-controller", "helm-controller", "notification-controller"},
+		ManifestFile: "flux-system.yaml",
+		Timeout:      5 * time.Second,
+	}
+
+	manifest, err := install.Generate(opts, "")
+	if err != nil {
+		return fmt.Errorf("couldn't generate manifests: %+v", err)
+	}
+
+	content := []byte(manifest.Content)
+
+	err = os.MkdirAll(filePath, os.ModePerm)
+	if err != nil {
+		return fmt.Errorf("couldn't create file %+v", err)
+	}
+
+	err = os.WriteFile(filepath.Join(filePath, shortFilename), content, 0666)
+	if err != nil {
+		return fmt.Errorf("couldn't write flux manifests to file %+v", err)
+	}
+
+	rf := rootFlags{
```

Comment:
Will remove everything related to root flags, copy-pasted and I think here this code is not needed, we already used called `MakeDefaultOptions` when generating the manifests.